src/main/java/com/google/devtools/build/remote/client/RemoteClient.java

```diff
@@ -407,5 +450,16 @@ public static void main(String[] args) throws Exception {
       return;
     }
 
+    if (optionsParser.getParsedCommand() == "run") {
+      Action.Builder builder = Action.newBuilder();
+      FileInputStream fin = new FileInputStream(runCommand.file);
+      TextFormat.getParser().merge(new InputStreamReader(fin), builder);
+      if (runCommand.path != null) {
```

Comment:
My personal preference:

```
client.setupDocker(
  builder.build(),
  runCommand.path != null ? runCommand.path : Files.createTempDir().toPath());
```

Up to you.

Reply:
Also done.